components/payments/settings/payment-webhooks.tsx

```diff
@@ -7,5 +21,210 @@ interface PaymentsWebhooksProps {
+        onError: () => {
+          triggerAlert(
+            "error",
+            "Failed to Create Webhook",
+            `Failed to create  ${
+              webhook.isProduction ? "production" : "testnet"
+            } webhook: ${webhook.url}`,
+          );
+        },
+      },
+    );
+  };
+
+  const onDeleteWebhook: PaymentsWebhooksTableProps["onDelete"] = (webhook) => {
+    // mutate
+    updateWebhook(
+      { webhookId: webhook.id, deletedAt: new Date() },
+      {
+        onSuccess: () => {
+          triggerAlert(
+            "success",
+            "Webhook Deleted",
+            `Successfully deleted ${
+              webhook.isProduction ? "production" : "testnet"
+            } webhook: ${webhook.url}`,
+          );
+        },
+        onError: () => {
+          triggerAlert(
+            "error",
+            "Failed to Delete Webhook",
+            `Failed to delete ${
+              webhook.isProduction ? "production" : "testnet"
+            } webhook: ${webhook.url}`,
+          );
+        },
+      },
+    );
+  };
+
+  const createWebhookHandlerFactory = (isProduction: boolean) => {
+    const onAddWebhook: PaymentsWebhooksTableProps["onCreate"] = async (
+      url,
+    ) => {
+      if (!isValidWebhookUrl(url)) {
+        triggerAlert(
+          "error",
+          "Invalid Webhook Url",
+          `${url} is not a valid webhook url, please try a different url`,
+        );
+        return;
+      }
+
+      // mutate
+      createWebhook(
+        { url: url, isProduction },
+        {
+          onSuccess: () => {
+            triggerAlert(
+              "success",
+              "Webhook Updated",
+              `Successfully updated ${
+                isProduction ? "production" : "testnet"
+              } webhook: ${url}`,
+            );
+          },
+          onError: () => {
+            triggerAlert(
+              "error",
+              "Failed to Update Webhook",
+              `Failed to update ${
+                isProduction ? "production" : "testnet"
+              } webhook to url: ${url}`,
+            );
+          },
+        },
+      );
+    };
+    return onAddWebhook;
+  };
+
+  const isLoading = isCreateLoading || isGetLoading || isUpdateLoading;
+
+  return (
+    <>
+      <Card
+        p={8}
+        as={Flex}
+        flexDir="column"
+        gap={8}
+        maxW={{ base: "full", xl: "70%" }}
+      >
+        <Flex flexDir="column" gap={2}>
+          <Heading>Webhooks</Heading>
+        </Flex>
+        <DetailsRow
+          title="Secret Key"
+          tooltip={`Used for authenticating the webhook request`}
+          content={
+            isLoadingSecretKey ? (
+              <Spinner size="sm" />
+            ) : (
+              <CodeBlock code={webhookApiKey?.data?.decrypted_key ?? ""} />
+            )
+          }
+        />
+
+        <Divider />
+
+        <Flex flexDir="column" gap={2}>
+          <Heading size={"title.md"}>Production</Heading>
```

Comment:
nit, always nice if you only have a string to not use brackets
```suggestion
          <Heading size="title.md">Production</Heading>
```